aws-observability/aws-application-signals-test-framework
# Check if ASG instances are ready before proceeding to validation

*Issue description:*
This PR aims to address 2 issues: 
1. EC2 Instance created by ASG was unhealthy ([Link](https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/10410519220/job/28832546846))
2. Traffic Generator took too long to become ready ([Link](https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/10405692731/job/28817032152))

Both cases are happening because we are proceeding to log validation immediately after we deploy the asg without checking whether the EC2 instances launched by the asg is ready or not.

*Description of changes:*
1. Added a step to wait till the EC2 instance status is 'ok' before proceeding to validation step. For the first issue, it takes around 12 minutes for the ASG to refresh unhealthy instances so the wait time was set to 15 minutes. This issue only occurs in EC2 ASG cases so it is acceptable to sometimes have the EC2 case go over 15 minutes as it will not block any subsequent runs. 
2. Reduced health grace period time to 3 minutes. This is to reduce the 12 minute mentioned above to 10 minute.
4. Increased log validation time from ~2.5 minutes to ~3.3 minutes. Change 1 usually takes around 1.5 minutes if there are no issues, so giving 2.3 more minutes for the traffic generator to become available is enough time to resolve issue 2. (Although this isn't really neccessary because change 1 should resolve the issue anyways, added it as a fail-safe)

*Test Run*:
- Tested in private account to check whether reducing the grace period to 3 minutes will cause any issue.
- Test [run](https://github.com/aws-observability/aws-application-signals-test-framework/actions/runs/10529856233) for ASG

*Rollback procedure:*
Revert PR

<Can we safely revert this commit if needed? If not, detail what must be done to safely revert and why it is needed.>

*Ensure you've run the following tests on your changes and include the link below:*

To do so, create a `test.yml` file with `name: Test` and workflow description to test your changes, then remove the file for your PR. Link your test run in your PR description. This process is a short term solution while we work on creating a staging environment for testing.

NOTE: TESTS RUNNING ON A SINGLE EKS CLUSTER CANNOT BE RUN IN PARALLEL. See the [needs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idneeds) keyword to run tests in succession.
- Run Java EKS on `e2e-playground` in us-east-1 and eu-central-2
- Run Python EKS on `e2e-playground` in us-east-1 and eu-central-2
- Run metric limiter on EKS cluster `e2e-playground` in us-east-1 and eu-central-2
- Run EC2 tests in all regions
- Run K8s on a separate K8s cluster (check IAD test account for master node endpoints; these will change as we create and destroy clusters for OS patching)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
